### DISCUSSION TITLE
Is there specific event for column header click?

### DISCUSSION THREAD

**Discussion by Ignalion** [bokeh/bokeh#13760] [General]
I have custom server side pagination made for DataTable, because data is very big and the DataTable itself does not support any pagination.
Now I need to somehow implement server side sorting for it now. The only thing I can think of is having callback for clicking the column header but I’m not sure it is possible.

The ideal way probably will be catch some sort of 'sorting event' and implement server side sorting, but I'm open to other possibilities. As far as I understand the sorting itself happens completely on the javascript side, so I have little hope about that. But if I could have some control over displaying/clicking the column header I could achieve the desired result.

**Comment by bryevdv** [comment]
@mattpap Do you happen to know offhand if latest SlickGrid exposes an event we could hook into? IIRC last I looked several years ago that was not really the case.

**Reply by mattpap** [reply]
There's `onHeaderClick` event handler available.

**Reply by Ignalion** [reply]
Is it python callback? How/when is it triggered? I cannot catch it with on_event() on datatable instance

**Reply by mattpap** [reply]
It's an underlying SlickGrid's API, which needs to be exposed via a Bokeh API.

**Reply by Ignalion** [reply]
Is there a solution or workaround for doing that?

**Reply by bryevdv** [reply]
@Ignalion No, my question was only to establish whether the underlying SlickGrid offered the necessary events for us to even be able to do something *in principle*. Now that it has be established that SlickGrid does offer the necessary events, then it is *possible* that *future* feature development work could be on Bokeh done to expose that capability. There is nothing available (that is useful to you as an end-user), right now.

**Comment by bryevdv** [comment]
And actually, there is still some more to investigate before I'd convert this to a feature request issue. Namely, in addition to hooking into the SlickGrid `onHeaderClick` can we also disable/surpress the default SlickGrid sorting? I assume that is the case, but we would need to make certain, since otherwise a header click event by itself would probabliy not be useful for your use case of custom sorting.

**Reply by GillesFa** [reply]
I have an app with the data of cds split in 2 datatables. I need to know when a user sorts a table so that this sorting can be replicated to the other table  
Can that event "onHeaderClick" (once exposed) be used for that?

**Comment by Ignalion** [comment]
Hey guys. Just wondering if there any news regarding the issue?
